controllers/object_controls.go

```diff
@@ -1327,6 +1327,11 @@ func transformForRuntime(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec,
 		setContainerEnv(mainContainer, "CONTAINERD_RUNTIME_CLASS", getRuntimeClass(config))
 	}
 
+	if runtime == gpuv1.CRIO.String() {
+		// We add the nvidia runtime to the cri-o config by default instead of installing the OCI prestart hook
+		setContainerEnv(mainContainer, CRIOConfigModeEnvName, "config")
```

Comment:
Do we now do this twice, or is the `mainContainer` different from `obj.Spec.Template.Spec.Containers[0]`?

Reply:
I have updated this so that we only set this envvar here.